plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1921,11 +1929,12 @@ public void finishStatisticsCollection(ConnectorSession session, ConnectorTableH
         updateProperties.commit();
 
         CollectedStatistics collectedStatistics = processComputedTableStatistics(table, computedStatistics);
+        boolean isFullAnalyze = handle.getAnalyzeHandle().map(IcebergAnalyzeHandle::isFullAnalyze).orElse(false);
         StatisticsFile statisticsFile = tableStatisticsWriter.writeStatisticsFile(
                 session,
                 table,
                 snapshotId,
-                REPLACE,
+                isFullAnalyze ? REPLACE : INCREMENTAL_UPDATE,
```

Comment:
what would happen if we used INCREMENTAL_UPDATE always?

Reply:
You would call unnecessarily call code that merges changes instead of just replacing them i think nothing else

Reply:
INCREMENTAL_UPDATE facilitates stats update during inserts
we cannot update stats during deletions though. That's why we need REPLACE mode. ANALYZE uses REPLACE mode to "erase" values that are no longer in the table. (this is a challenge for incremental analyze, but that's another story).

now, ANALYZE wants to use REPLACE mode regardless of whether all data columns got analyzed, or only the subset. Otherwise it wouldn't be able to fix stats drifting over time. see https://github.com/trinodb/trino/pull/16889

Reply:
I cherry-picked your pr and the tests passed so I'd assume we are fine. Or what am I missing ?

Reply:
That test doesn't cover the case where subset of columns is being analyzed. Can modify the test accordingly?

Reply:
I've updated https://github.com/trinodb/trino/pull/16889 to include analyze with a subset of columns.
The current version of this PR will break that test.